database-couchdb/index.js

```diff
@@ -0,0 +1,103 @@
+const fs = require('fs');
+const PATH = require('path');
+const csvParser = require('csv-parser');
+const Axios = require('axios');
+
+const availabilityPath = PATH.resolve(__dirname, '..', 'data-gen', 'csv', 'availability.csv');
+const storesPath = PATH.resolve(__dirname, '..', 'data-gen', 'csv', 'stores.csv');
+
+const startingItemId = 100;
+const batchSize = 100;
+
+const removeOldData = (url, db) => {
+
+  return Axios.delete(url + db)
+    .catch((err) => {
+      return err;
+    });
+};
+
+const createNewData = (url, db) => {
+  let curId = startingItemId;
+  let availabilityStream = fs.createReadStream(availabilityPath);
+
+  const writeData = async (curIds, stores) => {
+    let count = 0;
+    let docs = [];
+    let entry = {};
+    let curId = curIds
+    availabilityStream
+      .on('error', (err) => {
+        console.error(err);
+      })
+      .pipe(csvParser())
+      .on('data', (data) => {
+
+        count++;
+        if (entry.itemId === undefined) {
```

Comment:
What does this mean exactly?

Reply:
Basically after every 5 lines from the csv, i reset `entry` to be an empty object. If the key itemId doesnt exist, that means entry is ready to be repopulated with new information